File: C++/binarySearch.cpp
/*
 * Christopher Bennett
 * Binary Search in C++
 * 24/07/2024
*/


#include <iostream>

using namespace std;



int main()
{
	cout << "We Ball!" << "\n";
	cout << "Bianry Search\n";

	int start = 0;
	int arr[11] = {0,1,2,3,4,5,6,7,8,9,10};
	int arrSize = sizeof(arr) / sizeof(int);
	int end = arrSize;
	int mid = (start + end) / 2;
	int value;
	bool Found = false;
	
	cout << "Enter the number your looking for" << "\n";
	cin >> value; cout << "You Entered: " << value << "\n";


	while (start < end)
	{
		if (arr[mid] == value){
			cout << "\n\n\n\t"  << "value found at index: " << mid << "\n";
			cout << "\n\n\n";
			start = end + 1;
			Found = true;
		}else if (arr[mid] < value){
			start = mid + 1;
		}else if (arr[mid] > value){
			end = mid - 1;
		}

		mid = (start + end) / 2;
	}
	if (!Found){cout << "That number is not in the list :(" << "\n";}
	return 0;
}

File: Java/binarySearch.java
//
//Christopher Bennett
//Binary Search Algo




import java.util.Scanner;



public class binarySearch
{
	public static void main(String[] arg)
	{
		System.out.println("We Ball");	
		
		Scanner sc = new Scanner(System.in);
		boolean isFound = false ;
		double[] arr = {0,1,23,33,45,56,67,78,89,99};
		int size = arr.length;
		int start = 0;
		int end = size - 1;
		int mid = (start + end) / 2;

		//getting lost value
		System.out.println("Enter number to found");
		double value = sc.nextDouble();

		//Binary Search Algo	
		while (!isFound)
		{
			mid = (start + end) /2;
			if (start > end){
				System.out.println("Value is not in array");
				System.exit(0);
			}

			if (arr[mid] == value){
				System.out.println("Value Found at index " + mid);
				isFound = true;
				System.exit(0);
			}else if (arr[mid] < value){
				System.out.println("to the right");
				start = mid + 1;
			}else if (arr[mid] > value){
				System.out.println("to the left");
				end =  mid - 1;
			}
		}
		
	}
}

